### ISSUE TITLE
Correctly decompile interface methods with bodies

### ISSUE THREAD

**Issue by svick** [ashmind/SharpLab#227]
The Roslyn branch features/DefaultInterfaceImplementation can generate interfaces whose methods have bodies. Those show correctly when disassembled to IL, but not when decompiled to C#.

For example, consider this code:

```c#
interface IFoo
{
	int DefaultMethod() => 42;
}
```

[It is correctly disassembled to IL as:](https://sharplab.io/#v2:EYLgZgpghgLgrgJwgZwLQBEJinANjASQDsYIFsBjCAgWwAdcIaITYBLAeyIBoYQ3cAHzYkylCAAICAMQ4cAsACgA3koCQImBMzY8MALIQYACw4ATABQBKCQF4AfBIAsAJgDcSgL5A===)

```cil
.class interface private auto ansi abstract IFoo
{
    // Methods
    .method public hidebysig newslot virtual 
        instance int32 DefaultMethod () cil managed 
    {
        // Method begins at RVA 0x2050
        // Code size 3 (0x3)
        .maxstack 8

        IL_0000: ldc.i4.s 42
        IL_0002: ret
    } // end of method IFoo::DefaultMethod

} // end of class IFoo
```

[But decompiling to C# doesn't show the body:](https://sharplab.io/#v2:EYLgZgpghgLgrgJwgZwLQBEJinANjASQDsYIFsBjCAgWwAdcIaITYBLAeyIB82SzKEAAQEAYhw4BYAFABvGQEg+MIZmx4YAWQgwAFhwAmACgCUQgLwA+IQBYATAG4ZAXyA==)

```c#
internal interface IFoo
{
    int DefaultMethod();
}
```

As far as I can tell, the current version of ILSpy does support decompiling such methods: when I build Roslyn from that branch locally, compile the above code using the built csc and decompile it using ILSpy built from master using:

```c#
var decompiler = new CSharpDecompiler(assembly.MainModule, new DecompilerSettings());
Console.WriteLine(decompiler.DecompileWholeModuleAsString());
```

Then the output includes the method body:

```c#
using System.Diagnostics;
using System.Reflection;
using System.Runtime.CompilerServices;

[assembly: CompilationRelaxations(8)]
[assembly: RuntimeCompatibility(WrapNonExceptionThrows = true)]
[assembly: Debuggable(DebuggableAttribute.DebuggingModes.Default | DebuggableAttribute.DebuggingModes.DisableOptimizations | DebuggableAttribute.DebuggingModes.IgnoreSymbolStoreSequencePoints | DebuggableAttribute.DebuggingModes.EnableEditAndContinue)]
[assembly: AssemblyVersion("0.0.0.0")]
internal interface IFoo
{
        int DefaultMethod()
        {
                return 42;
        }
}
```

**Comment by ashmind** [comment]
@svick Thanks! I plan to move to latest ILSpy, however I would need to submit some of my changes to ILSpy first.

**Comment by ashmind** [comment]
Postponed until #231.

**Comment by ashmind** [comment]
Fixed by #238.
